src/Utils/Miscellaneous.php

```diff
@@ -20,4 +20,59 @@ public static function systemTime()
     {
         return intval(round(microtime(true) * 1000));
     }
+
+    private static function deepConvertObjectToArrayRecursive(&$item , $_key)
+    {
+        if(!is_array($item) && !is_object($item))
+            return;
+
+        $item = array_filter(
+            (array) $item,
+            static function (string $key): bool {
+                return strpos($key, "\0") !== 0;
+            }, ARRAY_FILTER_USE_KEY);
+
+        foreach($item as &$value){
+            if(is_object($value)) {
+                array_walk($value, [__CLASS__, __FUNCTION__]);
+            }
+        }
+    }
+
+    public static function deepConvertObjectToArray(&$object)
+    {
+        array_walk($object, [__CLASS__, 'deepConvertObjectToArrayRecursive']);
+        $object = (array)$object;
+    }
+
```

Comment:
Lots of double newlines in here, can we make the formatting consistent in this file